draft-ietf-dnsop-ns-revalidation.mkd

```diff
@@ -175,31 +177,33 @@ Attacks exploiting lack of this revalidation have been described in {{GHOST1}},
-    Positive responses to this validation query will be cached with an authoritative data ranking.
-    Successive queries directed to the same zone will be directed to the nameservers listed in the child's apex, due to the ranking of this answer.
-    If the validation query fails, the parent NS RRset will remain the one with the highest ranking and will be used for successive queries.
+A validation query consists of a query for the child's apex NS RRset, sent to the newly discovered delegation's nameservers.
+Normal iterative logic applies to the processing of responses to validation queries, including storing the results in cache, trying the next server on SERVFAIL or timeout, and so on.
+Positive responses to this validation query will be cached with an authoritative data ranking.
+Successive queries directed to the same zone will be directed to the nameservers listed in the child's apex, due to the ranking of this answer.
+If the validation query fails, the parent NS RRset will remain the one with the highest ranking and will be used for successive queries.
 
--   Additional validation queries for the "glue" resource records of referral responses (if not already authoritatively present in cache) may be sent with the validation query for the NS RRset as well.
-    Positive responses will be cached authoritatively and replace the non authoritative "glue" entries.
-    Successive queries directed to the same zone will be directed to the authoritative values for the names of the NS RRset in the referral response.
+When a resolver detects that the child's apex NS RRset contains different nameservers than the non-authoritative version at the parent side of the zone cut, it MAY report the mismatch using DNS Error Reporting {{RFC9567}} on the Report-Channel for the child zone, as well as on the Report-Channel for the parent zone, with an extended DNS error code of TBD {{IANA}}.
 
--   The names from the NS RRset from a validating NS query may differ from the names in NS RRset in the referral response.
-    Outstanding validation queries for "glue" that do not match names in the authoritative NS RRset be discarded, or they may be left running to completion.
-    Their result will no longer be used in queries for the zone.
-    Outstanding validation queries for "glue" that do match names in the authoritative NS RRset must be left running to completion.
-    They do not need to be re-queried after reception of the authoritative NS RRset (see [](#upgrade-addresses)).
+Additional validation queries for the "glue" resource records of referral responses (if not already authoritatively present in cache) may be sent with the validation query for the NS RRset as well.
+Positive responses will be cached authoritatively and replace the non authoritative "glue" entries.
+Successive queries directed to the same zone will be directed to the authoritative values for the names of the NS RRset in the referral response.
```

Comment:
i think "to the authoritative nameservers denoted in" sounds better than "to the authoritative values for the names of the NS RRset in".

Reply:
Agree. Resolved in [e306cc0](https://github.com/shuque/ns-revalidation/pull/26/commits/e306cc0fc3d20f94c378c37c56282973320259bc)